src/strkey.js

```diff
@@ -136,31 +137,110 @@ export class StrKey {
+  }
+
+  /**
+   * Encodes raw data to strkey signed payload (P...).
+   * @param   {Buffer} data  data to encode
+   * @returns {string}
+   */
+  static encodeSignedPayload(data) {
+    return encodeCheck('signedPayload', data);
+  }
+
+  /**
+   * Decodes strkey signed payload (P...) to raw data.
+   * @param   {string} address  address to decode
+   * @returns {Buffer}
+   */
+  static decodeSignedPayload(address) {
+    return decodeCheck('signedPayload', address);
+  }
+
+  /**
+   * Checks validity of alleged signed payload (P...) strkey address.
+   * @param   {string} address  signer key to check
+   * @returns {boolean}
+   */
+  static isValidSignedPayload(address) {
+    return isValid('signedPayload', address);
   }
 }
 
-// Warning: This isn't a *definitive* check of validity, but rather just a
-// basic-effort check.
+/**
+ * Sanity-checks whether or not a strkey *appears* valid.
+ *
+ * @param  {string}  versionByteName the type of strkey to expect in `encoded`
+ * @param  {string}  encoded         the strkey to validate
+ *
+ * @return {Boolean} whether or not the `encoded` strkey appears valid for the
+ *     `versionByteName` strkey type (see `versionBytes`, above).
+ *
+ * @note This isn't a *definitive* check of validity, but rather a best-effort
+ *     check based on (a) input length, (b) whether or not it can be decoded,
+ *     and (c) output length.
+ */
 function isValid(versionByteName, encoded) {
-  // it's either non-muxed && len=56, or muxed && len=69
-  if (encoded && encoded.length !== 56 && encoded.length !== 69) {
+  if (!isString(encoded)) {
     return false;
   }
 
-  try {
-    const decoded = decodeCheck(versionByteName, encoded);
-    if (decoded.length !== 32 && decoded.length !== 40) {
+  // basic length checks on the strkey lengths
+  switch (versionByteName) {
+    case 'ed25519PublicKey': // falls through
+    case 'ed25519SecretSeed': // falls through
+    case 'preAuthTx': // falls through
+    case 'sha256Hash':
+      if (encoded.length !== 56) {
+        return false;
+      }
+      break;
+
+    case 'med25519PublicKey':
+      if (encoded.length !== 69) {
+        return false;
+      }
+      break;
+
+    case 'signedPayload':
+      if (encoded.length < 56 || encoded.length > 165) {
+        return false;
+      }
+      break;
+
+    default:
       return false;
-    }
+  }
+
+  let decoded = '';
+  try {
+    decoded = decodeCheck(versionByteName, encoded);
   } catch (err) {
     return false;
   }
-  return true;
+
+  // basic length checks on the resulting buffer sizes
+  switch (versionByteName) {
+    case 'ed25519PublicKey': // falls through
+    case 'ed25519SecretSeed': // falls through
+    case 'preAuthTx': // falls through
+    case 'sha256Hash':
+      return decoded.length === 32;
+
+    case 'med25519PublicKey':
+      return decoded.length === 40;
+
+    case 'signedPayload':
+      return decoded.length >= 32 + 4 && decoded.length <= 32 + 4 + 64;
```

Comment:
My guess is that `4` is related to the length of the hint. Maybe a comment would be helpful to make this seem a little less magic number-y

Reply:
Yep, good call!